controllers/nstemplateset/namespaces_test.go

```diff
@@ -478,6 +480,89 @@ func TestEnsureNamespacesOK(t *testing.T) {
+		tests := []struct {
+			name            string
+			enabledFeatures string
+		}{
+			{
+				name:            "no enabled features",
+				enabledFeatures: "",
+			},
+			{
+				name:            "feature-1 enabled",
+				enabledFeatures: "feature-1",
+			},
+			{
+				name:            "feature-1 and feature-2 enabled",
+				enabledFeatures: "feature-1,feature-2",
+			},
+			{
+				name:            "feature-1 and feature-3 enabled",
+				enabledFeatures: "feature-1,feature-3",
+			},
+			{
+				name:            "feature-2 enabled",
+				enabledFeatures: "feature-2",
+			},
+			{
+				name:            "all features enabled",
+				enabledFeatures: "feature-1,feature-2,feature-3",
+			},
+		}
+		for _, testRun := range tests {
+			t.Run(testRun.name, func(t *testing.T) {
+				// given
+
+				// Create a NSTemplate referring to a tier with multiple RoleBinding objects
+				// The first one is with no feature annotation.
+				// The next three represent feature-1, feature-2 and feature-3.
+				allTierFeatures := []string{"feature-1", "feature-2", "feature-3"}
+				nsTmplSet := newNSTmplSet(namespaceName, spacename, "advanced", withNamespaces("abcde11", "dev", "stage"), withConditions(Provisioning()))
+				if testRun.enabledFeatures != "" {
+					nsTmplSet.Annotations = map[string]string{
+						toolchainv1alpha1.FeatureToggleNameAnnotationKey: testRun.enabledFeatures,
+					}
+				}
+				devNS := newNamespace("advanced", spacename, "dev", withTemplateRefUsingRevision("abcde11")) // NS exist but it is not complete yet
+				manager, fakeClient := prepareNamespacesManager(t, nsTmplSet, devNS)
+
+				// when
+				createdOrUpdated, err := manager.ensure(ctx, nsTmplSet)
+
+				// then
+				require.NoError(t, err)
+				assert.True(t, createdOrUpdated)
+				AssertThatNSTemplateSet(t, namespaceName, spacename, fakeClient).
+					HasFinalizer().
+					HasSpecNamespaces("dev", "stage").
+					HasConditions(Provisioning())
+
+				nsAssertion := AssertThatNamespace(t, spacename+"-dev", fakeClient).
+					HasLabel(toolchainv1alpha1.SpaceLabelKey, spacename).
+					HasLabel(toolchainv1alpha1.TypeLabelKey, "dev").
+					HasLabel(toolchainv1alpha1.TemplateRefLabelKey, "advanced-dev-abcde11").
+					HasLabel(toolchainv1alpha1.TierLabelKey, "advanced").
+					HasLabel(toolchainv1alpha1.ProviderLabelKey, toolchainv1alpha1.ProviderLabelValue).
+					HasResource("crtadmin-pods", &rbacv1.RoleBinding{})
+
+				// Check enabled feature resources
+				enabledFeatures := strings.Split(testRun.enabledFeatures, ",")
+				// Check that the rest of the resources for disabled features are not created
+				for _, feature := range allTierFeatures {
+					if slices.Contains(enabledFeatures, feature) {
+						nsAssertion.HasResource(fmt.Sprintf("%s-rb", feature), &rbacv1.RoleBinding{})
+					} else {
+						nsAssertion.HasNoResource(fmt.Sprintf("%s-rb", feature), &rbacv1.RoleBinding{})
+					}
+				}
+
+				AssertThatNamespace(t, spacename+"-stage", manager.Client).
```

Comment:
we could maybe add a comment here explaining why the stage namespace is not there. I guess it's because we provisioned only the `dev` namespace.

Reply:
Correct. We probably don't need this assertion at all because we don't care about the stage namespace in this test but anyway... I added the comment: https://github.com/codeready-toolchain/member-operator/pull/580/commits/4dbe286e0f90e4541e10878c21c7d94aba501db5